packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1261,6 +1263,8 @@ function prepareFreshStack(root, expirationTime) {
   workInProgressRootNextUnprocessedUpdateTime = NoWork;
   workInProgressRootHasPendingPing = false;
 
+  resetMutableSourceWorkInProgressVersions(isPrimaryRenderer);
```

Comment:
Let's move this to `completeWork` and `unwindWork` of HostRoot so it gets cleared when exiting the render phase, instead of when starting the next render phase that happens to follow that. (This is akin to popping context providers off the stack.)

Reply:
I think this suggested change (adb5a2f) causes problems for high-pri interruptions, since moving this would mean that the WIP version doesn't get reset between a low priority render and a flush sync.